src/AtomsBase.jl

```diff
@@ -0,0 +1,133 @@
+import AtomsBase: velocity, position, bounding_box, boundary_conditions, periodicity
+import AtomsBase: n_dimensions, atomic_mass, atomic_number, atomic_symbol, element
+
+"""
+    position(frame::Frame)
+    position(frame::Frame, index)
+
+This function is part of the AtomsBase.jl interface
+
+Return a vector of positions of every particle in the system `frame`. Return type
+should be a vector of vectors each containing `3` elements that are
+`<:Unitful.Length`. If an index is passed or the action is on an `Atom`,
+return only the position of the referenced `Atom` on that index.
+"""
+function position(frame::Frame)
+    cf_pos = positions(frame)
+    ab_pos = [cf_pos[:,i] for i in 1:size(cf_pos,2)]*u"Å"
+    return ab_pos
+end
+position(frame::Frame, index) = position(frame)[index] 
+#ISSUE position not saved in Atom 
```

Comment:
Instead of using `chemfiles.Atom` as the AtomsBase atom, you can define a new `Chemfiles.AtomsBaseAtom` which contains it index, the original chemfiles Atom and the original chemfiles Frame.

Reply:
That works for the most part. The issue I'm having right now is that getindex(frame::Frame, index) returns a Chemfiles.Atom object while the species type used by AtomsBase would be an AtomsBaseAtom. In AtomsBase getindex(system::AbstractSystem, index) usually returns the same data type used by AtomsBase functions. I'm not sure if that one is an explicit requirement, though.